os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c

```diff
@@ -851,13 +894,54 @@ _timerHandle _tizenrt_timerCreate(const signed char *pcTimerName, osdepTickType
 	timer->timevalue = xTimerPeriodInTicks;
 	timer->data = pvTimerID;
 	timer->function = pxCallbackFunction;
+
+	if (!_tizenrt_timer_table_init) {
+		INIT_LIST_HEAD(&_tizenrt_timer_table);
+		_tizenrt_timer_table_init = 1;
+	}
+
+	struct _tizenrt_timer_entry *timer_entry;
+	timer_entry = _tizenrt_zmalloc(sizeof(struct _tizenrt_timer_entry));
+	if (timer_entry == NULL) {
+		kmm_free(timer->work_hdl);
+		kmm_free(timer);
+		return NULL;
+	}
+	memset(timer_entry, 0, sizeof(struct _tizenrt_timer_entry));
+	timer_entry->timer = timer;
+
+	save_and_cli();
+	list_add(&(timer_entry->list), &_tizenrt_timer_table);
```

Comment:
Is it ok not ot protect list_add()?
Following scenario might cause crash please check
```
queue status:  tizenrt_timer_table -> 1 -> 2
1) task A call _tizenrt_timer_Create()
2) task B executes _tizenrt_timerDelete and remove 1 item in the list by list_del_init()
=> context switching pameters of __list_del are (tizenrt_timer_table, 2)
3) task A executes list_add to add 0 item to tizenrt_timer_table 
3-1) 1.prev = 0
3-2) 0.next = tizenrt_timer_table .next (1)
3-3) 0.prev = tizenrt_timer_table 
3-4) tizenrt_timer_table.next = 0
4) context switching task B runs __list_del
4-1) 2.prev = tizenrt_timer_table
4-2) tizenrt_timer_table.next = 2
```
Then item 0 might be missing. please check.

Reply:
save_and_cli() is calling irqsave(), so it will prevent context switching.
So list_add() is actually protected.

Reply:
Sorry I was confused. It's ok because list_add() is protected